### DISCUSSION TITLE
Permissions for guests

### DISCUSSION THREAD

**Discussion by kibcode** [codeigniter4/shield#751] [Ideas]
### PHP Version

8.1

### CodeIgniter4 Version

4.3.5

### Shield Version

1.0.0-beta.7

### Which operating systems have you tested for this bug?

Windows, Linux

### Which server did you use?

apache

### Database

MySQL

### Did you customize Shield?

No

### What happened?

When i want to check permissions with can() and the user is not logged in shield produces an error. I always have to check with loggedIn() before using can().

### Steps to Reproduce

Try to get a permission for guests.

### Expected Output

can() should return false if a user is not logged in. It would also be nice to handle guests as a usergroup so users not logged in can have permissions too.

**Comment by kibcode** [comment]
Is there a good workaround i have missed so far?

**Comment by kenjis** [comment]
This is not a bug.

`can()` is a method in `User`.
https://codeigniter4.github.io/shield/authorization/#can

When the user is not logged in, `auth()->user()` returns `null`.
So you cannot call `can()`.

**Reply by kibcode** [reply]
thanks for clearifying. yes, i know its not directly a bug and i got why it doesnt work but its an issue for me. lets say i have an items comment cell and want to disallow specific usergroups to view that cell and also disallow it for guest. that means i have to call loggedIn() every time before can(). is that correct or is there a better practice?
